### PR TITLE
Improve introspector 'mixed cluster' error messages... 

### PR DESCRIPTION
**Pull request by tbarnes-us**
… plus other minor changes to introspector.

### CODE REVIEW

File: operator/src/main/resources/scripts/introspectDomain.py [hunk 1]
```diff
@@ -575,19 +575,19 @@ def validateDynamicClusterReferencedByOneServerTemplate(self, cluster):
...
             return
     if server_template is None:
-      self.addError("The dynamic cluster " + self.name(cluster) + "' is not referenced by any server template.")
+      self.addError("The dynamic WebLogic cluster " + self.name(cluster) + "' is not referenced by any server template.")
 
   def validateDynamicClusterNotReferencedByAnyServers(self, cluster):
     for server in self.env.getDomain().getServers():
       if self.env.getClusterOrNone(server) is cluster:
-        self.addError("The dynamic cluster " + self.name(cluster) + " is referenced by the server " + self.name(server) + ".")
+        self.addError("The dynamic WebLogic cluster " + self.name(cluster) + " is referenced by configured server " + self.name(server) + ", the operator does not support 'mixed clusters' that host both dynamic (templated) servers and non-dynamic (configured) servers.")
```
Comment (tbarnes-us) [inline operator/src/main/resources/scripts/introspectDomain.py:586]: This is one of two mixed cluster checks ^^^.

File: operator/src/main/resources/scripts/introspectDomain.py [hunk 2]
```diff
@@ -448,12 +448,12 @@ def validateNonDynamicClusterReferencedByAtLeastOneServer(self, cluster):
...
       if self.env.getClusterOrNone(server) is cluster:
         return
-    self.addError("The non-dynamic cluster " + self.name(cluster) + " is not referenced by any servers.")
+    self.addError("The non-dynamic WebLogic cluster " + self.name(cluster) + " is not referenced by any servers.")
 
   def validateNonDynamicClusterNotReferencedByAnyServerTemplates(self, cluster):
     for template in self.env.getDomain().getServerTemplates():
       if self.env.getClusterOrNone(template) is cluster:
-        self.addError("The non-dynamic cluster " + self.name(cluster) + " is referenced by the server template " + self.name(template) + ".")
+        self.addError("The non-dynamic WebLogic cluster " + self.name(cluster) + " is referenced by the server template " + self.name(template) + ", the operator does not support 'mixed clusters' that host both dynamic (templated) servers and non-dynamic (configured) servers.")
```
Comment (tbarnes-us) [inline operator/src/main/resources/scripts/introspectDomain.py:456]: This is one of two 'mixed cluster' checks ^^^.

**Comment by tbarnes-us** [comment]
@alai8 wrote:> Should we also get feedback from Rosemary?

Not sure if she's ever looked at the introspector log messages before.  Couldn't hurt.  (I hope!)

File: operator/src/main/resources/scripts/introspectDomain.py [hunk 3]
```diff
@@ -434,7 +434,7 @@ def validateCluster(self, cluster):
       self.validateNonDynamicCluster(cluster)
     else:
       if self.env.isFMWInfraDomain() and not self.env.allowDynamicClusterInFMWInfraDomain():
-        self.addError("Dynamic clusters are not supported in FMW Infrastructure domains. Set ALLOW_DYNAMIC_CLUSTER_IN_FMW environment variable to true to bypass this validation.")
+        self.addError("Dynamic WebLogic clusters are not supported in FMW Infrastructure domains. Set ALLOW_DYNAMIC_CLUSTER_IN_FMW environment variable to true to bypass this validation.")
```
Comment (rosemarymarano) [inline operator/src/main/resources/scripts/introspectDomain.py:437]: The correct term here would be, "WebLogic dynamic clusters" ( as opposed to WebLogic configured clusters), not "Dynamic WebLogic clusters." (Please make this correction globally.)

File: operator/src/main/resources/scripts/introspectDomain.py [hunk 4]
```diff
@@ -448,12 +448,12 @@ def validateNonDynamicClusterReferencedByAtLeastOneServer(self, cluster):
     for server in self.env.getDomain().getServers():
       if self.env.getClusterOrNone(server) is cluster:
         return
-    self.addError("The non-dynamic cluster " + self.name(cluster) + " is not referenced by any servers.")
+    self.addError("The non-dynamic WebLogic cluster " + self.name(cluster) + " is not referenced by any servers.")
```
Comment (rosemarymarano) [inline operator/src/main/resources/scripts/introspectDomain.py:451]: The WLS documentation does not use the term "non-dynamic" cluster (and I don't think that we should either), only dynamic or configured, https://docs.oracle.com/en/middleware/standalone/weblogic-server/14.1.1.0/clust/dynamic_clusters.html#GUID-DA7F7FAD-49AA-4F3D-8A05-0D9921B96971.

File: operator/src/main/resources/scripts/introspectDomain.py (hunk 2 above)
Comment (rosemarymarano) [inline operator/src/main/resources/scripts/introspectDomain.py:456]: There is a concept of a "mixed cluster" (see docs), but not non-dynamic. (Please make this correction globally.)

Comment (tbarnes-us) [reply operator/src/main/resources/scripts/introspectDomain.py]: done

File: operator/src/main/resources/scripts/introspectDomain.py (hunk 4 above)
Comment (tbarnes-us) [reply operator/src/main/resources/scripts/introspectDomain.py]: done

File: operator/src/main/resources/scripts/introspectDomain.py (hunk 3 above)
Comment (tbarnes-us) [reply operator/src/main/resources/scripts/introspectDomain.py]: done

**Pull request merged**